### PR TITLE
clang-format: use thiserror for errors

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
This removed the enum from the public API.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -120,43 +120,32 @@ pub enum ClangFormatError {
...
-        } else {
-            Err(ClangFormatError::WaitFailure)
-        }
+    // Wait for the output and parse it
+    let output = child.wait_with_output()?;
+    // TODO: use exit_ok() once it is a stable feature
+    // https://doc.rust-lang.org/stable/std/process/struct.ExitStatus.html#method.exit_ok
+    // https://github.com/rust-lang/rust/issues/84908
+    if output.status.success() {
+        String::from_utf8(output.stdout).map_err(ClangFormatError::from)
```
Comment (LeonMatthesKDAB) [inline src/lib.rs:146]:
```suggestion
        Ok(String::from_utf8(output.stdout)?)
```

Hm, this is a very minor thing, but the current code I think reads a bit as if there is never "Ok" returned.
That's probably just me being used to seeing `Ok(...)` at the end of a function though.
The map_err approach is more functional as well, which is also nice.

**Review by LeonMatthesKDAB — DISMISSED** [review]
LGTM

File: src/lib.rs (hunk 1 above)
Comment (ahayzen-kdab) [reply src/lib.rs:146]: I think the Ok thing is slightly more readable what is happening and avoids the awkward `ClangFormatError::from`, so i'll change it.

**Review by LeonMatthesKDAB — APPROVED** [review]
Reapproved

**Pull request merged**